src/Function.cpp

```diff
@@ -331,6 +331,35 @@ Function::Function(const std::vector<Type> &required_types, int required_dims, c
     contents->required_dims = required_dims;
 }
 
+void Function::update_with_deserialization(const std::string &name, const std::string &origin_name, const std::vector<Halide::Type> &output_types,
```

Comment:
...wow, this has a lot of arguments.
(1) Code style nit: for this many args, I'd argue that we should really put one per line (both here and in the header), as it's hard to read and check, IMHO)
(2) I wonder if this would be better as a struct argument: done well, that would allow some of probably-lesser-used args (e.g. debug_file) to have default values that don't have to be filled in. OTOH, I guess the 'struct' would really be basically just the entire FunctionContents, which is intentionally hidden.

Maybe I'm overthinking point 2 above; I suspect this function will ~never be called by user code directly (only indirectly via the deserialization code).